### ISSUE TITLE
Unable to Invoke Spring Beans

### ISSUE THREAD

**Issue by Tapsarchitect** [decebals/pf4j#49]
Hi Team,

I have used yours Spring plugins with the PF4j. The thing is its working with the main apps, But its not working on the Plugins created on the Plugins.

Reason its looking into the plugin class loader which pf4j and but if its look into the Webclassloader then only I can have the spring context.

In the Main app I have 
-------
AnnotationConfigApplicationContext applicationBeanContext = new AnnotationConfigApplicationContext();
		applicationBeanContext.register(ResponseRegistryBootInjector.class);
		applicationBeanContext.refresh();
		this.applicationContext = applicationBeanContext;
------

In the client App we have

------
ApplicationContext context = (ApplicationContext) mspUiContextObj.getContextInfo("APPSKEY");
		Object obj = context.getBean(LoginFormListener.class);
------

While debug on the above last statement "context.getBean(LoginFormListener.class)", I saw that its taking from the pluginclassloader not from the Spring which is in my case Webclassloader.
Eg here it is from the pf4j pluginclassloader "@Override
    public Class<?> loadClass(String className) throws ClassNotFoundException {
        log.debug("Received request to load class '{}'", className);"

Please assist.

**Comment by decebals** [comment]
It's not clear for me what di you try to achieve. I understand that it's a problem related to [pf4j-spring] and not in pf4j.
I understand that you have a Spring web application and you want to use pf4j, pf4j-spring in your application. Do you have an error stack? What is LoginFormListener, it's an extension point ?
Please give me more information.

**Comment by Tapsarchitect** [comment]
I would like to make it little detail..
1. I have the Main Application , where I am loading the plugin classes using the pf4j. The main application has the pf4j-spring as well. Now If I write the spring services like @Component classes and beans classes it works well.

2. Now I have folder under the main application called modules. Inside these module folder I keep the plugin classes. now how these plugin classes look is as per your one of the demo.

3. Plugin class which is a separate jar , created using the extension point.

@Configuration
public class JSONResponsePlugin extends Plugin {
	
        @Autowired
	ResourceService resourceService; 
	
    public JSONResponsePlugin(PluginWrapper wrapper) {
		super(wrapper);
	}

    @Override
    public void start() {
    	AnnotationConfigApplicationContext applicationContext = new AnnotationConfigApplicationContext();
		applicationContext.register(JSONResponsePlugin.class);
		applicationContext.refresh();
		ResourceService resourceService = applicationContext.getBean(ResourceService.class);
		resourceService.getResourceModel();
    	// TODO Write your actions when the plugin is got loaded
    	System.out.println("---JSONResponsePlugin.start()");
        // for testing the development mode
        if (RuntimeMode.DEVELOPMENT.equals(wrapper.getRuntimeMode())) {
        	System.out.println(StringUtils.upperCase("ResponsePlugin"));
        }
    }

    @Override
    public void stop() {
        // TODO ON STOP 
    	System.out.println("---JSONResponsePlugin.stop()");
    }

    @Extension
    public class JSonResponse implements Response {

		@Override
		public String parseData() {
			// Get Your Spring Objects
			String data = resourceService.getResourceModel().getName();
			return data;
		}
		
    }
    
	@Bean
	public ResourceService resourceService()
	{
		return new ResourceService();
	}

}

-----
Now Just wanted to clear one thing on you spring demo project, I see that if you put under the same demo spring classes it works.

But I initiated through extension point classes I cannot see these spring beans are coming up.
Can you share any demo example which is written using @Extension point and there you have autowired the spring components.

Thanks,
Tapan Pratap

**Comment by decebals** [comment]
http://stackoverflow.com/questions/25682399/spring-data-exception-furing-mongorepository-init help you?

**Comment by decebals** [comment]
I don't understand this construction in your `Plugin.start` method:
```java
AnnotationConfigApplicationContext applicationContext = new AnnotationConfigApplicationContext();
applicationContext.register(JSONResponsePlugin.class);
applicationContext.refresh();
ResourceService resourceService = applicationContext.getBean(ResourceService.class);
resourceService.getResourceModel();
```

In `pf4j-spring` I created an application context in `Boot/Main` (only in this class) and I read all extensions for an extension point only with an `@Autowired` (see `Grettings` class).  I don't know if you can put a spring annotation in your `Plugin` class because the plugin is instantiated by PF4J, it's not a spring bean. I think that it's ok to put spring annotation in your extension but I must improve the code to resolve the spring annotations from extensions (now it's not possible).

I will improve the pf4j-spring to show you the code.

So, in conclusion, you want to put some spring annotation in your extension. Correct?

**Comment by Tapsarchitect** [comment]
Hi,

Yes, you are correct, Currently its initiated by the Pf4j classloader not the Spring classloader.
So, as you mention, please help to improve the code and write the demo for the Spring based classes I mean annotations from the extension classes.

Thanks,
Tapan Pratap.

**Comment by decebals** [comment]
Moved to decebals/pf4j-spring/issues/2
